src/s-core-devcontainer/.devcontainer/setup-proxy.sh

```diff
@@ -0,0 +1,218 @@
+            fi
+            echo "Waiting for Docker daemon... ($i/30)"
+            sleep 2
+        done
+
+        # Verify Docker is working
+        if ! docker info > /dev/null 2>&1; then
+            echo "❌ Failed to start Docker daemon, trying alternative approach..."
+
+            # Try without config file
+            sudo pkill dockerd > /dev/null 2>&1 || true
+            sleep 2
+            sudo HTTP_PROXY="${HTTP_PROXY:-}" HTTPS_PROXY="${HTTPS_PROXY:-}" \
+                 http_proxy="${http_proxy:-}" https_proxy="${https_proxy:-}" \
+                 NO_PROXY="${NO_PROXY:-}" no_proxy="${no_proxy:-}" \
+                 dockerd > /dev/null 2>&1 &
+
+            sleep 5
+            if ! docker info > /dev/null 2>&1; then
+                echo "❌ Failed to start Docker daemon"
+                return 1
+            fi
+        fi
+
+        # Verify proxy settings are applied
+        if docker info | grep -q "HTTP Proxy\|HTTPS Proxy"; then
+            echo "✅ Proxy settings successfully applied to Docker daemon"
+        else
+            echo "⚠️  Proxy settings may not be fully applied, but Docker is working"
+        fi
+    }
+
+    # Restart Docker with proxy settings
+    restart_docker_with_proxy
+
+    # Configure BuildKit with proxy
+    echo "🏗️  Configuring BuildKit with proxy settings..."
+
+    # Remove existing multiarch builder if it exists
+    docker buildx rm multiarch 2>/dev/null || true
+
+    # Prepare buildx command with proxy options
+    BUILDX_CMD="docker buildx create --name multiarch --driver docker-container --driver-opt network=host"
+
+    # Add proxy environment variables only if they are set and non-empty
+    if [[ -n "${HTTP_PROXY:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.HTTP_PROXY=${HTTP_PROXY}"
+    fi
+
+    if [[ -n "${HTTPS_PROXY:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.HTTPS_PROXY=${HTTPS_PROXY}"
+    fi
+
+    if [[ -n "${http_proxy:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.http_proxy=${http_proxy}"
+    fi
+
+    if [[ -n "${https_proxy:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.https_proxy=${https_proxy}"
+    fi
+
+    # Add NO_PROXY only if set
+    if [[ -n "${NO_PROXY:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.NO_PROXY=${NO_PROXY}"
+    fi
+
+    if [[ -n "${no_proxy:-}" ]]; then
+        BUILDX_CMD="$BUILDX_CMD --driver-opt env.no_proxy=${no_proxy}"
+    fi
+
+    # Execute the buildx command
+    eval $BUILDX_CMD
+
+    # Use the new builder
+    docker buildx use multiarch
+
+    # Bootstrap the builder
+    echo "🚀 Bootstrapping BuildKit instance..."
+    docker buildx inspect --bootstrap
+
+    echo "✅ Docker and BuildKit configured with proxy settings"
+
+else
+    echo "🌐 No proxy environment detected, using default Docker configuration..."
+
+    # Start Docker daemon if not running
+    if ! pgrep dockerd > /dev/null; then
+        echo "🔄 Starting Docker daemon..."
```

Comment:
nice, I actually had a system once where I needed to always manually start the daemon. Super annoying.